etcdserver/server.go

```diff
@@ -466,11 +470,24 @@ func NewServer(cfg *ServerConfig) (srv *EtcdServer, err error) {
 // modify a server's fields after it has been sent to Start.
 // It also starts a goroutine to publish its server information.
 func (s *EtcdServer) Start() {
+	s.wg.Add(4)
```

Comment:
this is error-prone. it is easy to forget add one when we add a new go routine in start(). can we just do add(1) before each of the go keyword?